### PR TITLE
Switch from react-app-rewired to regular CRA

### PR DESCRIPTION
**Pull request by RDIL**
🎉 
fyi @geekflyer 

this pr is sponsored by vim, the only editor cli I know how to use
yeah no not really

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,21 +49,19 @@
...
-    "build": "react-app-rewired build",
+    "start": "react-scripts start",
+    "start-untyped": "TSC_COMPILE_ON_ERROR=\"true\" react-scripts start",
+    "build": "react-scripts build",
     "relay": "npx relay-compiler --verbose --watchman false --src ./src --schema schema.graphql --language typescript",
     "sync-schema": "get-graphql-schema https://api.cirrus-ci.com/schema.json > schema.graphql",
-    "test": "react-app-rewired test --env=jsdom",
+    "test": "react-scripts test --env=jsdom",
     "typecheck": "tsc -p tsconfig.json --noEmit",
-    "eject": "react-app-rewired eject",
```
Comment (RDIL) [inline package.json]: Hopefully this should never need to be ejected

**Comment by RDIL** [comment]
For plugin compatibility reasons, I did have to upgrade all dependencies because without it there were issues

File: src/cirrusTheme.ts [hunk 2]
```diff
@@ -56,15 +57,9 @@ let cirrusTheme = {
       hover: cirrusColors.cirrusGrey,
     },
   },
-  overrides: {
```
Comment (RDIL) [inline src/cirrusTheme.ts]: https://github.com/mui-org/material-ui/pull/17584 broke this so need to find another solution

**Comment by fkorotkov** [comment]
Were you able to run it locally? I'm getting the following warning in the console and nothing loads:

>react-dom.development.js:12487 Warning: componentWillMount has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.
>
>* Move code with side effects to componentDidMount, and set initial state in the constructor.
>* Rename componentWillMount to UNSAFE_componentWillMount to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.
>
>Please update the following components: LoadableComponent

**Comment by RDIL** [comment]
Thats been a thing for a while now, also I'm currently on a chromebook :(

**Comment by fkorotkov** [comment]
How did you tested the change then?

**Comment by RDIL** [comment]
I'm trying to ... haven't found a good site online for it

**Comment by fkorotkov** [comment]
OK, so this PR is not ready for a review then. Please ping me once it's ready. 🙌

**Comment by RDIL** [comment]
This is why I don't like unpinned dependencies 😞

**Comment by RDIL** [comment]
So I think I fixed the broken updating of MUI, so can you please give it another shot @fkorotkov?
Oh also `componentWillUnmount` has been deprecated since March.

**Comment by fkorotkov** [comment]
Thank you for doing this! @RDIL, you rock!

**Comment by RDIL** [comment]
Haha that's why I own https://rdil.rocks

**Comment by fkorotkov** [comment]
That's why I phrased that way ;-)

**Pull request merged**